### PR TITLE
hash passwords on save

### LINKED ISSUE #420: Refactor Hash Password Architecture
**Purpose**
What will this feature provide?

Simplify the current architecture we have for storing/hashing passwords. 

**Proposed Solution**
If you have an idea on how to implement this feature, please share!

Currently hashing methods are stored in /utils/auth.py and are imported to /resources/user.py to be used for the register, login, and patch endpoints. It also has to be imported to multiple files that deal with testing user endpoints.

To simplify it - the methods can be migrated directly into the User model, so hashing and checking can occur in one place.

**Acceptance Criteria**
Please list the requirements the implementation of this feature should meet.

N/A

**Mockup Screenshot or Page Number**
Paste a screenshot of the feature from the FIGMA mockups or cite the page number.

N/A

**Additional context**

None

### PR DESCRIPTION
**Pull request by NickSchimek**
### What issue is this solving?

Closes codeforpdx/dwellingly-app/issues/420

### Any helpful knowledge/context for the reviewer?
Is a re-seeding of the database necessary? YES
Any new dependencies to install? No
Any special requirements to test? No

### CODE REVIEW

File: models/base_model.py [hunk 1]
```diff
@@ -32,8 +32,7 @@ def create(cls, schema, payload):
             abort(400, err.messages)
 
         obj = cls(**attrs)
-        db.session.add(obj)
```
Comment (cecillesalazar) [inline models/base_model.py:35]: Nice work drying this up here~

**Review by cecillesalazar — APPROVED** [review]
Looks good. I tested manually too - just checked through print statements. Any test we make will probably not be very explicit, so I'm OK with keeping things as is if you are 👍

File: models/base_model.py (hunk 1 above)
Comment (NickSchimek) [reply models/base_model.py:35]: This change was actually necessary which highlights the fact that I should have a test for this.

File: resources/user.py [hunk 2]
```diff
@@ -155,7 +153,7 @@ def post(self):
         if user and user.archived:
             return {"message": "Invalid user"}, 403
 
-        if user and check_pw(data['password'], user.password):
+        if user.check_pw(data['password']):
```
Comment (NickSchimek) [inline resources/user.py:156]: It seems like this should be `if user and user.check_pw(...)` - Created ticket codeforpdx/dwellingly-app/issues/427 to deal with this.

File: models/base_model.py (hunk 1 above)
Comment (NickSchimek) [reply models/base_model.py:35]: I will address in this ticket https://github.com/codeforpdx/dwellingly-app/issues/426

**Pull request merged**